### PR TITLE
Fix scrolling and layout issues on All results page

### LINKED ISSUE #832: Problems with scrolling on All results page
At least when scrolling sideways, the table headings (and filter cells) of the sticky columns don't stay still and the how-many-to-show-on-a-page, paginations, buttons for downloading, etc. also move and hide. All of those should probably stick.

kuva
kuva

For some reason, this was not fixed in the PR #797

### LINKED ISSUE #845: Results view: sticky column headers messed up when scrolling horizontally
On "All results" view the headers for left-hand sticky columns for name and student number are messed up when the table is scrolled horizontally. Not a major showstopper, but annoying and ugly, and should be fixed.

### PR DESCRIPTION
**Pull request by etanttila**
# Description

**What?**

Fix scrolling and layout issues (e.g. width of sticky columns) on All results page (scrolling and sticky columns don't work perfectly on small screens, but proved too difficult to fix).
Also show tooltips on summary row headers to improve ease of use. Also updated tooltip texts in the JSON-file to use the term "assignment" rather than "exercise", which will be soon changed elsewhere as well. Tooltips on row headers probably don't work on mobile devices.

**Why?**

Previously when scrolling, cells overlapped strangely and sticky cells didn't behave as wanted.

**How?**

CSS-styling; Bootstrap tooltips

Fixes #832
Fixes #845 (duplicate)

# Testing

**Remember to add or update unit tests for new features and changes.**

* How to [test your changes in A-plus](https://github.com/apluslms/a-plus/tree/master/doc#running-tests-and-updating-translations)
* How to [test accessibility](https://wiki.aalto.fi/display/EDIT/How+to+check+the+accessibility+of+pull+requests)

**What type of test did you run?**

- [ ] Accessibility test using the [WAVE](https://wave.webaim.org/extension/) extension.
- [ ] Django unit tests.
- [ ] Selenium tests.
- [ ] Other test. *(Add a description below)*
- [x] Manual testing.

Tested scrolling vertically and horizontally with different screen sizes. And tested that tooltips work on hover on a PC.
All points page is not very accessible, it has many contrast errors, problems with labels, etc. and these accessibility errors are not fixed in this PR.

**Did you test the changes in**

- [x] Chrome
- [x] Firefox
- [ ] This pull request cannot be tested in the browser.

**Think of what is affected by these changes and could become broken**

# Translation

- [x] Did you modify or add new strings in the user interface? ([Read about how to create translation](https://github.com/apluslms/a-plus/tree/master/doc#running-tests-and-updating-translations))

# Programming style

- [x] Did you follow our [style guides](https://apluslms.github.io/contribute/styleguides/)?
However, the indentation in the JS-file used indentation of 4 spaces, which is why I kept that.

# Have you updated the README or other relevant documentation?

- [ ] documents inside the doc directory.
- [ ] README.md.
- [ ] Aplus Manual.
- [ ] Other documentation (mention below which documentation).

# Is it Done?

- [ ] Reviewer has finished the code review
- [ ] After the review, the developer has made changes accordingly
- [ ] Customer/Teacher has accepted the implementation of the feature

*Clean up your git commit history before submitting the pull request!*

### CODE REVIEW

File: assets/js-translations/teacher.fi.json [hunk 1]
```diff
@@ -54,7 +54,7 @@
   "Organisaatio",
 
   "Student ID":
-  "Opiskelijanumero",
+  "Opiskelija\u00ADnumero",
```
Comment (PasiSa) [inline assets/js-translations/teacher.fi.json:57]: "Opiskelijanumero" is without hyphen

**Review by PasiSa — CHANGES_REQUESTED** [review]
Looks good to me. Spotted one small language issue though.

File: assets/js-translations/teacher.fi.json (hunk 1 above)
Comment (etanttila) [reply assets/js-translations/teacher.fi.json:57]: `\u00AD` is a soft hyphen or a syllable hyphen, so if the word fits on a single line, it's not there, but if it doesn't it appears with the word break. (Basically, this is a way to get the cell to be less wide and still follow grammar rules.)

Comment (PasiSa) [reply assets/js-translations/teacher.fi.json:57]: ok, then everything ok.

**Pull request merged**